engine/server/engine/server/enclave_rest_api_handler.go

```diff
@@ -0,0 +1,857 @@
+	enclave_identifier := request.EnclaveIdentifier
+	apiContainerClient, responseErr := manager.GetApiClientOrResponseError(enclave_identifier)
+	if responseErr != nil {
+		return api.PostEnclavesEnclaveIdentifierStarlarkPackagesdefaultJSONResponse{Body: *responseErr, StatusCode: int(responseErr.Code)}, nil
+	}
+	logrus.Infof("Upload Starlark package on enclave %s", enclave_identifier)
+
+	for {
+		// Get next part (file) from the the multipart POST request
+		part, err := request.Body.NextPart()
+		if err == io.EOF {
+			break
+		}
+		filename := part.FileName()
+		client, err := (*apiContainerClient).UploadStarlarkPackage(ctx)
+		if err != nil {
+			logrus.Errorf("Can't upload Starlark package using gRPC call with enclave %s, error: %s", enclave_identifier, err)
+			return nil, stacktrace.NewError("Can't upload Starlark package using gRPC call with enclave %s", enclave_identifier)
+		}
+		clientStream := grpc_file_streaming.NewClientStream[rpc_api.StreamedDataChunk, emptypb.Empty](client)
+
+		_, err = clientStream.SendData(
+			filename,
+			part,
+			0, // Length unknown head of time
+			func(previousChunkHash string, contentChunk []byte) (*rpc_api.StreamedDataChunk, error) {
+				return &rpc_api.StreamedDataChunk{
+					Data:              contentChunk,
+					PreviousChunkHash: previousChunkHash,
+					Metadata: &rpc_api.DataChunkMetadata{
+						Name: filename,
+					},
+				}, nil
+			},
+		)
+		if err != nil {
+			// TODO(edgar) Should we stop on failure in case of multiple files? Should we return a list of succeed uploads?
+			return nil, err
+		}
+	}
+
+	return api.PostEnclavesEnclaveIdentifierStarlarkPackages200Response{}, nil
+}
+
+// (POST /enclaves/{enclave_identifier}/starlark/packages/{package_id})
+func (manager *enclaveRuntime) PostEnclavesEnclaveIdentifierStarlarkPackagesPackageId(ctx context.Context, request api.PostEnclavesEnclaveIdentifierStarlarkPackagesPackageIdRequestObject) (api.PostEnclavesEnclaveIdentifierStarlarkPackagesPackageIdResponseObject, error) {
+	enclave_identifier := request.EnclaveIdentifier
+	apiContainerClient, responseErr := manager.GetApiClientOrResponseError(enclave_identifier)
+	if responseErr != nil {
+		return api.PostEnclavesEnclaveIdentifierStarlarkPackagesPackageIddefaultJSONResponse{Body: *responseErr, StatusCode: int(responseErr.Code)}, nil
+	}
+	logrus.Infof("Run Starlark package on enclave %s", enclave_identifier)
+
+	package_id := request.PackageId
+	flags := utils.MapList(utils.DerefWith(request.Body.ExperimentalFeatures, []api_type.KurtosisFeatureFlag{}), toGrpcFeatureFlag)
+	// The gRPC always expect a JSON object even though it's marked as optional, so we need to default to `{}``
+	jsonParams := utils.DerefWith(request.Body.Params, map[string]interface{}{})
+	jsonBlob, err := json.Marshal(jsonParams)
+	if err != nil {
+		panic("Failed to serialize parameters")
```

Comment:
shouldn't be better to return a server error instead of panicking here? panic will stop the server I think

Reply:
IMO panicking is fine if you really don't expect that to fail, which is the case here. We are panicking here because we are serializing a deserialized JSON parameter and that never should fail. Also panicking here doesn't bring the server down, it only brings crashes the go routine handling that HTTP request. That said, I don't see any big issue on adding a bit of more code to pass the error back to the handler.